### PR TITLE
Adds a run-plugin operator to HRW

### PR DESCRIPTION
**Pull request by zwoop**
An example use case is to conditionally enable / run the rate_limit.so plugin for some requests:

```
cond %{REMAP_PSEUDO_HOOK} [AND]
cond %{CLIENT-HEADER:X-Leif} ="yes"
     run-plugin rate_limit.so "--limit=300 --error=429"
```

### CODE REVIEW

File: doc/admin-guide/plugins/header_rewrite.en.rst [hunk 1]
```diff
@@ -775,6 +775,14 @@ changing the remapped destination, ``<part>`` should be used to indicate the
...
 parts for rm-destination are QUERY, PATH, and PORT.
 
+run-plugin
+~~~~~~~~~~~~~~
+::
+
+  run-plugin <plugin-name>.so "<plugin-argument> ..."
+
+This allows to run an existing remap plugin, conditionally, from within a
+header rewrite rule.
```
Comment (ywkaras) [inline doc/admin-guide/plugins/header_rewrite.en.rst:785]: Can this run on any hook other than the remap hook?

**Comment by ywkaras** [comment]
Testing schmesting I guess.

**Review by shukitchan — DISMISSED** [review]

**Comment by shukitchan** [comment]
actually there is something wrong with the test.

but basically with the plugin factory, you can achieve this anywhere else, like Lua plugin or txn_box

**Comment by ywkaras** [comment]
I'll be on vacation for 2 weeks after today, so I'll I have to leave any further review to others.  Looks OK from a quick read through.

File: doc/admin-guide/plugins/header_rewrite.en.rst (hunk 1 above)
Comment (zwoop) [reply doc/admin-guide/plugins/header_rewrite.en.rst:785]: No it can only run in a remap hook, but the plugin it runs can setup other hooks.

**Comment by zwoop** [comment]
[approve ci]

**Comment by zwoop** [comment]
> Testing schmesting I guess.

Yeh, well... I can't reproduce the build problem on either my Mac or  my Linux box, but seems like it may be that the header_rewrite_test target needs a score dependency?

**Comment by zwoop** [comment]
[approve ci]

File: plugins/header_rewrite/operators.cc [hunk 2]
```diff
@@ -1103,3 +1106,73 @@ OperatorSetHttpCntl::exec(const Resources &res) const
...
+  require_resources(RSRC_CLIENT_REQUEST_HEADERS); // Need this for the txnp
+}
+
+void
+OperatorRunPlugin::exec(const Resources &res) const
+{
+  TSReleaseAssert(_plugin != nullptr);
+
+  if (res._rri && res.txnp) {
+    _plugin->doRemap(res.txnp, res._rri);
```
Comment (shukitchan) [inline plugins/header_rewrite/operators.cc:1176]: Just curious. should we pass on the return status of the doRemap() function somewhere?

Comment (zwoop) [reply plugins/header_rewrite/operators.cc:1176]: Yeh, so we kinda butchered this a bit in all the refactoring and dealings with "chaining" remap plugins. In most cases, this return code is obsoleted, and we don't' really use it. So, I made a choice to ignore it in HRW, whereas in Cripts, the caller can chose to look at it and do something with it (albeit, I think that's unlikely).

I'm not sure the stop / continue works any more either, but almost no plugin that's in use uses that.

**Review by shukitchan — APPROVED** [review]

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**